[ERROR] turicreate.toolkits._main: Toolkit error: Column type of target 'rating' must be int or str.

Hi, I am getting this error. 
```
[ERROR] turicreate.toolkits._main: Toolkit error: Column type of target 'rating' must be int or str.
Traceback (most recent call last):
  File "imdb_sent_turi.py", line 45, in <module>
    model = tc.sentence_classifier.create(dataset, 'rating', features=['text'])
  File "/Users/airphoenix/Desktop/aclImdb/sent_turi/lib/python3.6/site-packages/turicreate/toolkits/sentence_classifier/_sentence_classifier.py", line 110, in create
    validation_set=validation_set)
  File "/Users/airphoenix/Desktop/aclImdb/sent_turi/lib/python3.6/site-packages/turicreate/toolkits/classifier/logistic_classifier.py", line 312, in create
    seed=seed)
  File "/Users/airphoenix/Desktop/aclImdb/sent_turi/lib/python3.6/site-packages/turicreate/toolkits/_supervised_learning.py", line 398, in create
    options, verbose)
  File "/Users/airphoenix/Desktop/aclImdb/sent_turi/lib/python3.6/site-packages/turicreate/toolkits/_main.py", line 84, in run
    raise ToolkitError(str(message))
turicreate.toolkits._main.ToolkitError: Column type of target 'rating' must be int or str.
```

My ratings vector is derived from
```or pf in positiveFiles:
    with open(pf, "r") as f:
        indexCounter = 0
        review = f.read()
        split_title = pf.split('_')
        split_title2 = split_title[1].split('.')
        rating = int(split_title2[0])

        ratings_vector.append(rating)
        reviews_vector.append(review)`

```
and is, when printed, an array of ints.
My code for the model is:
```
dataset = {'rating': ratings_vector, 'text': reviews_vector}
model = tc.sentence_classifier.create(dataset, 'rating', features=['text'])
```

--- a/src/unity/python/turicreate/test/test_logistic_classifier.py
+++ b/src/unity/python/turicreate/test/test_logistic_classifier.py
@@ -8,8 +8,6 @@ from __future__ import division as _
 from __future__ import absolute_import as _
 import unittest
 import turicreate as tc
-import sys
-import operator as op
 import scipy.stats as ss
 import uuid
 import numpy as np
@@ -24,9 +22,6 @@ import os
 import copy
 import pandas as pd
 
-
-import os as _os
-_lfs = _os.environ['LFS_ROOT']
 
 #
 # Various test cases for the _LogisticRegressionClassifierModelTest

--- a/src/unity/python/turicreate/test/test_sentence_classifier.py
+++ b/src/unity/python/turicreate/test/test_sentence_classifier.py
@@ -6,10 +6,12 @@
 from __future__ import print_function as _
 from __future__ import division as _
 from __future__ import absolute_import as _
+
 import unittest
 import tempfile
 import coremltools
 import turicreate as tc
+from turicreate.toolkits._main import ToolkitError
 from turicreate.toolkits._internal_utils import _mac_ver
 from . import util as test_util
 
@@ -202,6 +204,17 @@ class SentenceClassifierCreateTests(unittest.TestCase):
         preds = m.classify(self.data)
         self.assertTrue(isinstance(preds, tc.SFrame))
         self.assertEqual(preds.column_names(), ["class", "probability"])
+
+    def test_not_sframe_create_error(self):
+        dataset = {'rating': [1,5], 'text': ['this is bad', 'this is good']}
+        try:
+            # dataset is NOT an SFrame
+            tc.sentence_classifier.create(dataset, 'rating', features=['text'])
+        except ToolkitError as t:
+            exception_msg = t.args[0]
+            self.assertTrue(exception_msg.startswith('Input dataset is not an SFrame. '))
+        else:
+            self.fail("This should have thrown an exception")
 
 
 class SentenceClassifierCreateBadValues(unittest.TestCase):

--- a/src/unity/python/turicreate/toolkits/sentence_classifier/_sentence_classifier.py
+++ b/src/unity/python/turicreate/toolkits/sentence_classifier/_sentence_classifier.py
@@ -6,7 +6,9 @@
 from __future__ import print_function as _
 from __future__ import division as _
 from __future__ import absolute_import as _
+
 import turicreate as _tc
+from turicreate.toolkits._internal_utils import _raise_error_if_not_sframe
 from turicreate.toolkits._model import CustomModel as _CustomModel
 from turicreate.toolkits._model import PythonProxy as _PythonProxy
 from turicreate.toolkits._internal_utils import _toolkit_repr_print
@@ -79,7 +81,7 @@ def create(dataset, target, features=None, method='auto', validation_set='auto')
 
     >>> metrics = m.evaluate(dataset)
     """
-    logger = _logging.getLogger(__name__)
+    _raise_error_if_not_sframe(dataset, "dataset")
 
     # Validate method.
     if method == 'auto':

